ddtrace/internal/import_hooks.py

```
@@ -0,0 +1,361 @@
+
+    # Update after we have successfully patched
+    _patched = True
+
+
+# DEV: This is called at the end of this module to ensure we always patch
+def patch():
+    """
+    Patch Python import system, enabling import hooks
+    """
+    # This should never cause their application to not load
+    try:
+        _patch()
+    except Exception:
+        log.debug('Failed to patch module importing, import hooks will not work', exc_info=True)
+
+
+def _unpatch():
+    # Only patch once
+    global _patched
+    if not _patched:
+        return
+    _patched = False
+
+    # 3.4 -> 3.8
+    # DEV: Explicitly stop at 3.8 in case the functions we are patching change in any way,
+    #      we need to validate them before adding support here
+    if (3, 4) <= sys.version_info <= (3, 8):
+        import importlib
+
+        if isinstance(importlib._bootstrap._find_and_load_unlocked, wrapt.FunctionWrapper):
+            setattr(
+                importlib._bootstrap, '_find_and_load_unlocked',
+                importlib._bootstrap._find_and_load_unlocked.__wrapped__,
+            )
+        if isinstance(importlib.reload, wrapt.FunctionWrapper):
+            setattr(importlib, 'reload', importlib.reload.__wrapped__)
+
+    # 2.7
+    # DEV: Slightly more direct approach
+    elif sys.version_info >= (2, 7):
+        if isinstance(__builtins__['__import__'], wrapt.FunctionWrapper):
+            __builtins__['__import__'] = __builtins__['__import__'].__wrapped__
+        if isinstance(__builtins__['reload'], wrapt.FunctionWrapper):
+            __builtins__['reload'] = __builtins__['reload'].__wrapped__
+
+
+def unpatch():
+    """
+    Unpatch Python import system, disabling import hooks
+    """
+    # This should never cause their application to not load
+    try:
+        _unpatch()
+    except Exception:
+        log.debug('Failed to unpatch module importing', exc_info=True)
```

It is really safe to just log?
I'd just let the exception be propagated, I mean, it's not like things are going to work well if `_unpatch` failed :(

---

I think letting the exception raise here is fine